include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -1623,19 +1644,17 @@ namespace internal
     n_q_points_1d,
     Number>::evaluate(const unsigned int                     n_components,
                       const EvaluationFlags::EvaluationFlags evaluation_flag,
-                      const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
-                      const Number *                                values_dofs,
-                      Number *                                      values_quad,
-                      Number *gradients_quad,
-                      Number *hessians_quad,
-                      Number *)
+                      const Number *                         values_dofs,
+                      EvaluationData &                       eval)
   {
+    const auto &shape_data = eval.get_shape_info().data[0];
+
     Assert(n_q_points_1d > fe_degree,
            ExcMessage("You lose information when going to a collocation space "
                       "of lower degree, so the evaluation results would be "
                       "wrong. Thus, this class does not permit the desired "
                       "operation."));
-    constexpr unsigned int n_q_points = Utilities::pow(n_q_points_1d, dim);
+    constexpr std::size_t n_q_points = Utilities::pow(n_q_points_1d, dim);
```

Comment:
Symmetry with other places. I will soon push an update that makes this more uniform in the file. The point is, as far as I can understand the generated code from compilers, that for addresses computed with `std::size_t` the compiler assumes that the index never overflows, hence allowing some optimizations not possible with `unsigned int` (on typical 64 bit systems), similar to https://github.com/dealii/dealii/blob/bf1a5c06d7d45a39f6347410989f9d2e47b15d5d/include/deal.II/matrix_free/fe_evaluation.h#L4742

Note that things do not matter here because we have compile-time constants for which the compiler knows that no overflow will happen.